// restful-mongoose/dist/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
	value: true
});

exports.default = function (name, model) {
	return (0, _resourceRouterMiddleware2.default)({
		id: name,

		list: function list(_ref, res) {
			var params = _ref.params;

			var limit = Math.max(1, Math.min(50, params.limit | 0 || 10));

			// if you have fulltext search enabled.
			if (params.search && typeof model.textSearch === 'function') {
				return model.textSearch(params.search, {
					limit: limit,
					language: 'en',
					lean: true
				}, (0, _util.toRes)(res));
			}

			model.find({}).skip(params.start | 0 || 0).limit(limit).exec((0, _util.toRes)(res));
		},
		create: function create(_ref2, res) {
			var body = _ref2.body;

			model.create(body, (0, _util.toRes)(res));
		},
		read: function read(_ref3, res) {
			var params = _ref3.params;

			model.findById(params[name], (0, _util.toRes)(res));
		},
		update: function update(_ref4, res) {
			var body = _ref4.body,
			    params = _ref4.params;

			delete body._id;
			model.findByIdAndUpdate(params[name], { $set: body }, (0, _util.toRes)(res));
		},
		delete: function _delete(_ref5, res) {
			var params = _ref5.params;

			model.findByIdAndRemove(params[name], (0, _util.toRes)(res));
		}
	});
};

var _resourceRouterMiddleware = require('resource-router-middleware');

var _resourceRouterMiddleware2 = _interopRequireDefault(_resourceRouterMiddleware);

var _util = require('./util');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

module.exports = exports['default'];
